### PR TITLE
Introduce `httpclient` for sending requests in DorisStreamLoad

### PR DESCRIPTION
**Pull request by bowenliang123**
# Proposed changes

- Refactor DorisStreamLoad to use `httpclient` for handling HTTP requests and repsonses

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @hf200012 @JNSimba

File: spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java [hunk 1]
```diff
@@ -247,39 +251,30 @@ private LoadResponse loadBatch(String value) {
...
+            httpPut.setEntity(new StringEntity(value, StandardCharsets.UTF_8));
+            HttpResponse httpResponse = httpClient.execute(httpPut);
+            responseHttpStatus = httpResponse.getStatusLine().getStatusCode();
+            String respMsg = httpResponse.getStatusLine().getReasonPhrase();
+            HttpEntity httpEntity = httpResponse.getEntity();
+            if (httpEntity != null) {
+                String response = EntityUtils.toString(new BufferedHttpEntity(httpEntity), StandardCharsets.UTF_8);
+                return new LoadResponse(responseHttpStatus, respMsg, response);
+            } else {
+                return new LoadResponse(responseHttpStatus, "", "");
```
Comment (JNSimba) [inline spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java:271]: Don't need to return `respMsg` here?

Comment (bowenliang123) [reply spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java]: Addressed.

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**